kong/tools/wrpc/proto.lua

```diff
@@ -0,0 +1,164 @@
+local pb = require "pb"
+local grpc = require "kong.tools.grpc"
+
+local grpc_new = grpc.new
+local pb_encode = pb.encode
+local setmetatable = setmetatable
+local string_format = string.format
+
+local _M = {}
+local _MT = { __index = _M, }
+
+local wrpc_proto_name = "wrpc.wrpc"
+
+local default_proto_path = { "kong/include/", "/usr/include/", }
+
+local function parse_annotation(annotation)
+  local parsed = {}
+  for kv_pair in annotation:gmatch("[^;]+=[^;]+") do
+    local key, value = kv_pair:match("^%s*(%S-)=(%S+)%s*$")
+    parsed[key] = value
+  end
+
+  return parsed
+end
+
+---@TODO: better way to do this
+-- Parse annotations in proto files with format:
+-- +wrpc: key1=val1; key2=val2; ...
+-- use key service-id and rpc-id to get IDs for service and rpc
+function _M:parse_annotations(proto_f)
```

Comment:
Unless we want to call parse_annotations from the exterior, I would not make this a public method, but instead a private function.

It is probably possible to make it return the annotations instead of modifying self.annotations:

```
self.annotations = parse_annotations(proto_f)
```

Reply:
It also saves some status to the proto object. But yes, this should not be exposed.